Fix the editor save keyboard shortcut not working in code editor view

talldan opened the pull request:
## Description
Something I spotted in my travels around Gutenberg—saving using the shortcut (Cmd+S/Ctrl+S) only works in the Visual Editor view, when switching to the CodeEditor view it ceases to work.

This PR extracts the save keyboard shortcut into a component `EditorSaveKeyboardShortcut`. That shortcut is now implemented in `TextEditorGlobalKeyboardShortcuts` (a new component) and `VisualEditorGlobalKeyboardShortcuts` (which has replaced `EditorGlobalKeyboardShortcuts`).

The old `EditorGlobalKeyboardShortcuts` component has been deprecated.

## How has this been tested?
1. Create a new post
2. Add a title and some content
3. Use the save shortcut (Cmd+S on a Mac, Ctrl + S elsewhere).
4. Observe that the post is saved
5. Switch to the Code Editor
6. Make some changes to the content
7. Use the save shortcut again
8. Observe that the post is saved (works in this branch, fails in master).
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

I tested the deprecation by temporarily replacing `VisualEditorGlobalKeyboardShortcuts` with `EditorGlobalKeyboardShortcuts` — the functionality should continue to work and the deprecation message should be displayed in the console.

## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
Bug fix (non-breaking change which fixes an issue)
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

talldan committed:
Fix the editor save keyboard shortcut not working in code editor view

noisysocks commented:
It's not unlikely that we'll want to add more global keyboard shortcuts to the text editor. Should we rename `EditorGlobalKeyboardShortcuts` to `VisualEditorGlobalKeyboardShortcuts` and `EditorSaveKeyboardShortcut` to `TextEditorGlobalKeyboardShortcuts`?

talldan commented:
> It's not unlikely that we'll want to add more global keyboard shortcuts to the text editor. Should we rename EditorGlobalKeyboardShortcuts to VisualEditorGlobalKeyboardShortcuts and EditorSaveKeyboardShortcut to TextEditorGlobalKeyboardShortcuts?

@noisysocks Maybe. I thought about it when working on this. `EditorSaveKeyboardShortcut` would still need to be a separate component to be usable in both `TextEditorGlobalKeyboardShortcuts` and `VisualEditorGlobalKeyboardShortcuts`. It's one of those things that has pros and cons on both sides, and I decided that maybe it's best not to introduce it while it's not needed.

talldan committed:
Refactor based on code review feedback

- Introduce new TextEditorGlobalShortcuts component that implements the save shortcut
- Rename EditorGlobalShortcuts to VisualEditorGlobalShortcuts
- Add a deprecated version of EditorGlobalShortcuts
- Update CHANGELOG

talldan commented:
I've refactored this to introduce `TextEditorGlobalKeyboardShortcuts` and `VisualEditorGlobalKeyboardShortcuts`.

noisysocks commented on packages/editor/src/components/global-keyboard-shortcuts/visual-editor-shortcuts.js:
```diff
@@ -169,35 +160,32 @@ export default compose( [
+export function EditorGlobalKeyboardShortcuts() {
+	deprecated( 'EditorGlobalKeyboardShortcuts', {
+		alternative: 'VisualEditorGlobalKeyboardShortcuts',
+		plugin: 'Gutenberg',
```
I wonder if we should still be using `plugin` now that Gutenberg and WP Core are essentially the same codebase?

(We do this elsewhere, so just a question 🙂)

noisysocks reviewed (approved):
:shipit:
